django/contrib/gis/geos/geometry.py

```diff
@@ -715,3 +651,68 @@ def closed(self):
         Return whether or not this Geometry is closed.
         """
         return capi.geos_isclosed(self.ptr)
+
+
+@deconstructible
+class GEOSGeometry(GEOSGeometryBase, ListMixin):
+    "A class that, generally, encapsulates a GEOS geometry."
+
+    def __init__(self, geo_input, srid=None):
+        """
+        The base constructor for GEOS geometry objects, and may take the
+        following inputs:
+
+         * strings:
+            - WKT
+            - HEXEWKB (a PostGIS-specific canonical form)
+            - GeoJSON (requires GDAL)
+         * buffer:
+            - WKB
+
+        The `srid` keyword is used to specify the Source Reference Identifier
+        (SRID) number for this Geometry.  If not set, the SRID will be None.
+        """
+        input_srid = None
+        if isinstance(geo_input, bytes):
+            geo_input = force_text(geo_input)
+        if isinstance(geo_input, str):
+            wkt_m = wkt_regex.match(geo_input)
+            if wkt_m:
+                # Handling WKT input.
```

Comment:
You could update the verbs tense to the convention used elsewhere: "Handle", "set", etc. rather than the "ing" versions.